File: lab2.go
package main

import "fmt"

func main() {
	var N int
	fmt.Scan(&N)

	var prev, curr int
	fmt.Scan(&prev)

	count := 0

	for i := 1; i < N; i++ {
		fmt.Scan(&curr)
		if curr == prev {
			count++
		}
		prev = curr
	}

	fmt.Printf("Результат: %d\n", count)
}

File: lab2.kt
import java.util.Scanner

fun main() {
    val scanner = Scanner(System.`in`)
    val N = scanner.nextInt()

    var prev = scanner.nextInt()
    var count = 0

    for (i in 1 until N) {
        val curr = scanner.nextInt()
        if (curr == prev) {
            count++
        }
        prev = curr
    }

    println("Результат: $count")
}